### PR TITLE
fix: Improve wording and flow of Tracing docs

### PR DESCRIPTION
**Pull request by bennothommo**
I was working with a custom Trace recently and found some parts of the Tracing docs a bit hard to follow or incomplete. This PR proposes some changes to the wording and flow to make it a little bit easier to read, fixed up a couple of spelling and grammatical errors and includes a few extra examples.

Happy to adjust anything - just let me know :)

### CODE REVIEW

**Comment by bennothommo** [comment]
@cla-bot check

**Review by NathanWalker — APPROVED** [review]
Really clear improvements here thank you!
